### ISSUE TITLE
SOCKS proxy connection does not use passed username/password

### ISSUE THREAD

**Issue by rahiel** [python-telegram-bot/python-telegram-bot#1087]
When setting the `https_proxy` environment variable like:

``` shell
export https_proxy=socks5://user:pass@IP:PORT
```
Then the vendored urllib3 does not parse and use the provided `user` and `pass`.

Fix: merge https://github.com/urllib3/urllib3/pull/1363 into the vendored urllib3 package.

(Originally reported at https://github.com/rahiel/telegram-send/issues/31)

**Comment by tsnoam** [comment]
@rahiel This upstream commit is not part of an official urllib3 release, which makes me consider it unstable.
This issue also has a workaround (programmatically setting the proxy instead of through an environment variable).

Regardless I'm not eager to add changes to the vendored urllib3 unless it is absolutely necessary.
I'm opening this up for comments from the community and the other maintainers.

**Comment by tsnoam** [comment]
The reason why I'm not thrilled to modify the vendored urllib3 is that it took us quite long to reach stability with this integration and I prefer not to rock the boat unless it's really necessary.

**Comment by Badiboy** [comment]
@rahiel use urllib3 kwargs. It works for me.
```
    request_kwargs={'proxy_url': 'socks5://xx.xx.xx.xx:yyyy/', 
                    'urllib3_proxy_kwargs': {"username": "user", "password": "pass"}}
    updater = Updater('zzzzzzzzzzzzzzzzz', workers = 10, request_kwargs = request_kwargs)

```

**Comment by rahiel** [comment]
@Badiboy the issue is with the `https_proxy` environment variable not working properly.

@iskhomutov there is no additional code needed in this library, as the fix already exists in urllib3.

**Comment by Badiboy** [comment]
@rahiel there is no "https_proxy environment variable" in my solution.

**Comment by rahiel** [comment]
> @rahiel This upstream commit is not part of an official urllib3 release, which makes me consider it unstable.
> This issue also has a workaround (programmatically setting the proxy instead of through an environment variable).
> 
> Regardless I'm not eager to add changes to the vendored urllib3 unless it is absolutely necessary.
> I'm opening this up for comments from the community and the other maintainers.

The upstream commit from urllib3 that would fix this issue, is now part of urllib3 [version 1.23](https://github.com/urllib3/urllib3/releases). So this fix can now be considered stable.

The issue now has also received 8 "thumb ups", so there is interest from the community to have this fixed.

**Comment by rahiel** [comment]
This was fixed with #1211.
